controllers/imageupdateautomation_controller.go

```diff
@@ -179,6 +181,19 @@ func (r *ImageUpdateAutomationReconciler) Reconcile(ctx context.Context, req ctr
 	}
 	debuglog.Info("fetching git repository", "gitrepository", originName)
 
+	if r.NoCrossNamespaceRef && auto.Spec.SourceRef.Namespace != auto.GetNamespace() {
```

Comment:
`auto.Spec.SourceRef.Namespace` can be empty, and this will reject the object even if it's not a cross-namespace ref. 

```suggestion
	if r.NoCrossNamespaceRef && gitRepoNamespace != auto.GetNamespace() {
```